cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/util/FileUtils.java

```diff
@@ -0,0 +1,119 @@
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * This utility class provides a more easy to use interface for handling
+ * YAML files requested from a local file source or a remote host source.
+ *
+ */
+public class FileUtils {
+
+    private static final Set<String> ALLOWED_FILE_EXTENSIONS = new HashSet<>(Arrays.asList(
+            "YAML",
+            "YML"
+    ));
+
+    /**
+     *
+     * @param filePath  a relative file path beginning from the working directory
+     * @return          the content of the file as a String
+     * @throws IOException
+     */
+    public static String readLocalFile(String filePath) throws IOException {
+        checkNotNull(filePath);
+
+        String content = doReadLocalFile(filePath);
+
+        assert content != null;
+        return content;
+    }
+
+    private static String doReadLocalFile(String filePath) throws IOException {
+        File file = new File(filePath);
+        if (!isYamlFile(file.getName())) {
+            throw new InvalidFileTypeException("invalid or missing file extension: "
+                    + FilenameUtils.getExtension(file.getName()));
+        }
+        return readFile(file);
+    }
+
+    /**
+     *
+     * @param url   url to the host e.g. https://host.com/path/to/file.yml
+     * @return      the content of the file as a String
+     * @throws IOException
+     */
+    public static String readRemoteFile(String url) throws IOException, ProtocolException {
+        checkNotNull(url);
+
+        String content = doReadRemoteFile(url);
+
+        assert content != null;
+        return content;
+    }
+
+    private static String doReadRemoteFile(String url) throws IOException, ProtocolException {
+        URI uri = URI.create(url);
+
+        if (!isYamlFile(uri.getPath()) && !emptyFileExtension(uri.getPath())) {
+            throw new InvalidFileTypeException("invalid file extension: "
+                    + FilenameUtils.getExtension(uri.getPath()));
+        }
+
+        try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+
+            try (CloseableHttpResponse response = httpClient.execute(new HttpGet(uri))) {
+
+                //TODO: more precise exception handling
+                if (response.getCode() != 200) {
+                    throw new HttpResponseException(response.getCode(), response.getReasonPhrase());
+                }
+
+                if (response.getHeader("Content-Type") != null
+                        && !response.getHeader("Content-Type").equals("text/plain")) {
+
+                    throw new HttpResponseException(response.getCode(),
+                            "invalid content type, was: " + response.getHeader("Content-Type"));
+                }
+
+                return EntityUtils.toString(response.getEntity());
+            }
+        }
+    }
+
+    public static boolean isYamlFile(String filename) {
```

Comment:
Docs missing on public methods.

Reply:
The method is not used outside the class, so we could just set it to private for now.